apps/airdrop/src/App/hooks.ts

```diff
@@ -0,0 +1,211 @@
+  const [_co, setConfirmation] = useAtom(confirmationAtom);
+
+  const handleMetamask = useCallback(async () => {
+    if (!metamask) {
+      handleExtensionDownload("https://metamask.io/download/");
+    } else {
+      await metamask.request({
+        method: "wallet_switchEthereumChain",
+        params: [{ chainId }],
+      });
+
+      const addresses = await metamask.request<string[]>({
+        method: "eth_requestAccounts",
+      });
+
+      if (!addresses || !addresses[0]) {
+        throw new Error("No address found");
+      }
+      const address = addresses[0];
+
+      let response: AirdropResponse<ExtensionClaim> | undefined;
+      try {
+        response = await checkExtensionClaim(address, "gitcoin");
+      } catch (e) {
+        console.error(e);
+      }
+      if (!response) {
+        toast("Something went wrong, please try again later");
+        return;
+      } else if (!response.ok) {
+        toast(`Something went wrong: ${response.result.message}`);
+        return;
+      }
+      const { result } = response;
+
+      let signature;
+      if (result.eligible && !result.has_claimed) {
+        signature = await metamask.request<string>({
+          method: "personal_sign",
+          params: [result.nonce, address],
+        });
+
+        if (!signature) {
+          throw new Error("Can't sign message");
+        }
+      } else if (result.eligible && result.has_claimed) {
+        setConfirmation({
+          confirmed: true,
+          address: result.airdrop_address as string,
+          publicKey: result.airdrop_public_key as string,
+          amount: result.amount,
+        });
+      }
+
+      setClaim({
+        eligible: result.eligible,
+        amount: result.amount,
+        hasClaimed: result.has_claimed,
+        signature,
+        address,
+        nonce: result.nonce,
+        type: "gitcoin" as const,
+      });
+
+      navigatePostCheck(navigate, result.eligible, result.has_claimed);
+    }
+  }, [metamask]);
+
+  return handleMetamask;
+};
+
+export const useKeplrHandler = (
+  chainId: string,
+  type: KeplrClaimType,
+  keplr?: Keplr
+): (() => Promise<void>) => {
+  const navigate = useNavigate();
+  const [_cl, setClaim] = useAtom(claimAtom);
+  const [_co, setConfirmation] = useAtom(confirmationAtom);
+
+  const handleKeplr = useCallback(async () => {
+    if (!keplr) {
+      handleExtensionDownload("https://www.keplr.app/download");
+    } else {
+      try {
+        await keplr.enable(chainId);
+      } catch (e) {
+        toast("Please allow Keplr to connect to this website");
+        return;
+      }
+
+      const { bech32Address: address } = await keplr.getKey(chainId);
+
+      let response: AirdropResponse<ExtensionClaim> | undefined;
+      try {
+        response = await checkExtensionClaim(address, type);
+      } catch (e) {
+        console.error(e);
+      }
+
+      if (!response) {
+        toast("Something went wrong, please try again later");
```

Comment:
Also not a big deal, but maybe an `enum` with constants of possible toast responses would be more maintainable, defined in a single place